File: ALODWebUtility.CS/Common/WebControlSetters.cs
using System;
using System.Web;
using System.Web.UI.WebControls;

namespace ALODWebUtility.Common
{
    public static class WebControlSetters
    {
        public static void SetCheckBox(CheckBox checkbox, bool? val)
        {
            if (val == null || !val.HasValue)
            {
                return;
            }

            checkbox.Checked = val.Value;
        }

        public static void SetDateLabel(Label label, DateTime? theDate)
        {
            if (label == null || !theDate.HasValue)
            {
                return;
            }

            SetLabelText(label, theDate.Value.ToString(Utility.DATE_FORMAT));
        }

        public static void SetDateTextbox(TextBox textbox, DateTime? theDate)
        {
            if (textbox == null || !theDate.HasValue)
            {
                return;
            }

            SetTextboxText(textbox, theDate.Value.ToString(Utility.DATE_FORMAT));
        }

        public static void SetDateTimeLabel(Label label, DateTime? dateTime)
        {
            if (label == null || !dateTime.HasValue)
            {
                return;
            }

            SetLabelText(label, dateTime.Value.ToString(Utility.DATE_HOUR_FORMAT));
        }

        public static void SetDropdownByValue(DropDownList list, string value)
        {
            if (list == null || string.IsNullOrEmpty(value))
            {
                return;
            }

            if (list.Items.FindByValue(value) != null)
            {
                list.SelectedValue = value;
            }
        }

        public static void SetLabelText(Label label, string text)
        {
            if (label == null || string.IsNullOrEmpty(text))
            {
                return;
            }

            label.Text = HttpContext.Current.Server.HtmlDecode(text);
        }

        public static void SetMaxLength(TextBox input)
        {
            input.Attributes.Add("maxLength", input.MaxLength.ToString());
            Utility.AddCssClass(input, "textLimit");
        }

        public static void SetMaxLength(TextBox input, bool charRemaining)
        {
            input.Attributes.Add("maxLength", input.MaxLength.ToString());
            if (charRemaining)
            {
                Utility.AddCssClass(input, "textLimit");
            }
        }

        public static void SetTextboxText(TextBox textbox, string text)
        {
            if (textbox == null || string.IsNullOrEmpty(text))
            {
                return;
            }

            textbox.Text = HttpContext.Current.Server.HtmlDecode(text);
        }

        public static void SetTextboxText(TextBox textbox, decimal? text)
        {
            if (textbox == null || !text.HasValue)
            {
                return;
            }

            textbox.Text = HttpContext.Current.Server.HtmlDecode(text.Value.ToString());
        }

        public static void SetTextboxText(TextBox textbox, int? text)
        {
            if (textbox == null || !text.HasValue)
            {
                return;
            }

            textbox.Text = HttpContext.Current.Server.HtmlDecode(text.Value.ToString());
        }

        public static void SetTimeTextbox(TextBox textbox, DateTime? timeAsDateTime)
        {
            if (textbox == null || !timeAsDateTime.HasValue)
            {
                return;
            }

            SetTextboxText(textbox, timeAsDateTime.Value.ToString(Utility.HOUR_FORMAT));
        }
    }
}

File: ALODWebUtility/Common/WebControlSetters.vb
Imports System.Web
Imports System.Web.UI.WebControls

Namespace Common
    Public Module WebControlSetters

        Public Sub SetCheckBox(ByVal checkbox As CheckBox, ByVal val As Boolean?)
            If (val Is Nothing OrElse Not val.HasValue) Then
                Exit Sub
            End If

            checkbox.Checked = val
        End Sub

        Public Sub SetDateLabel(ByVal label As Label, ByVal theDate As Nullable(Of DateTime))
            If (label Is Nothing OrElse Not theDate.HasValue) Then
                Exit Sub
            End If

            SetLabelText(label, theDate.Value.ToString(DATE_FORMAT))
        End Sub

        Public Sub SetDateTextbox(ByVal textbox As TextBox, ByVal theDate As Nullable(Of DateTime))
            If (textbox Is Nothing OrElse Not theDate.HasValue) Then
                Exit Sub
            End If

            SetTextboxText(textbox, theDate.Value.ToString(DATE_FORMAT))
        End Sub

        Public Sub SetDateTimeLabel(ByVal label As Label, ByVal dateTime As Nullable(Of DateTime))
            If (label Is Nothing OrElse Not dateTime.HasValue) Then
                Exit Sub
            End If

            SetLabelText(label, dateTime.Value.ToString(DATE_HOUR_FORMAT))
        End Sub

        Public Sub SetDropdownByValue(ByVal list As DropDownList, ByVal value As String)
            If (list Is Nothing OrElse String.IsNullOrEmpty(value)) Then
                Exit Sub
            End If

            If (list.Items.FindByValue(value) IsNot Nothing) Then
                list.SelectedValue = value
            End If
        End Sub

        Public Sub SetLabelText(ByVal label As Label, ByVal text As String)
            If (label Is Nothing OrElse String.IsNullOrEmpty(text)) Then
                Exit Sub
            End If

            label.Text = HttpContext.Current.Server.HtmlDecode(text)
        End Sub

        Public Sub SetMaxLength(ByVal input As TextBox)
            input.Attributes.Add("maxLength", input.MaxLength.ToString())
            AddCssClass(input, "textLimit")
        End Sub

        Public Sub SetMaxLength(ByVal input As TextBox, ByVal charRemaining As Boolean)
            input.Attributes.Add("maxLength", input.MaxLength.ToString())
            If (charRemaining) Then
                AddCssClass(input, "textLimit")
            End If

        End Sub

        Public Sub SetTextboxText(ByVal textbox As TextBox, ByVal text As String)
            If (textbox Is Nothing OrElse String.IsNullOrEmpty(text)) Then
                Exit Sub
            End If

            textbox.Text = HttpContext.Current.Server.HtmlDecode(text)
        End Sub

        Public Sub SetTextboxText(ByVal textbox As TextBox, ByVal text As Nullable(Of Decimal))
            If (textbox Is Nothing OrElse Not text.HasValue) Then
                Exit Sub
            End If

            textbox.Text = HttpContext.Current.Server.HtmlDecode(text)
        End Sub

        Public Sub SetTextboxText(ByVal textbox As TextBox, ByVal text As Nullable(Of Integer))
            If (textbox Is Nothing OrElse Not text.HasValue) Then
                Exit Sub
            End If

            textbox.Text = HttpContext.Current.Server.HtmlDecode(text)
        End Sub

        Public Sub SetTimeTextbox(ByVal textbox As TextBox, ByVal timeAsDateTime As Nullable(Of DateTime))
            If (textbox Is Nothing OrElse Not timeAsDateTime.HasValue) Then
                Exit Sub
            End If

            SetTextboxText(textbox, timeAsDateTime.Value.ToString(HOUR_FORMAT))
        End Sub

    End Module
End Namespace